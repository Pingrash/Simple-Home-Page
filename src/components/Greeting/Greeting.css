.greeting-container {
  flex: 1;
  margin: 2rem auto;
}

.greeting-body {
  width: 100%;
  margin: 0 auto;
  text-transform: capitalize;
  text-align: center;
  font-size: 3rem;
}
