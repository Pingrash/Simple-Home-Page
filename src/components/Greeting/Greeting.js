import React from 'react';
import PropTypes from 'prop-types';

import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  greetingContainer: {
    flex: 1,
    margin: '2rem auto'
  },
  greetingBody: {
    width: '100%',
    margin: '0 auto',
    textTransform: 'capitalize',
    textAlign: 'center'
  }
}));

const Greeting = ({ username, timeGreeting }) => {
  const classes = useStyles();

  return (
    <div className={classes.greetingContainer} data-test='container'>
      <Typography
        component='h2'
        variant='h2'
        className={classes.greetingBody}
        data-test='greeting'
      >
        Good {timeGreeting}
        {username ? (
          <span data-test='username'> {username}!</span>
        ) : (
          '!'
        )}
      </Typography>
    </div>
  );
};

Greeting.propTypes = {
  username: PropTypes.string,
  timeGreeting: PropTypes.string.isRequired
};

export default Greeting;
