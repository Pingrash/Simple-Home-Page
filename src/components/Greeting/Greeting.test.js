import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Greeting from './Greeting';
import { findByDataTest, checkProps } from '../../../Utilities';

// Snapshot check
describe('Greeting snapshot checks', () => {
  it('Renders correctly without props', () => {
    const tree = renderer.create(<Greeting />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly with props', () => {
    const tree = renderer
      .create(<Greeting username='user' timeGreeting='morning' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const setup = (props = {}) => {
  const component = shallow(<Greeting {...props} />);
  return component;
};

describe('Greeting component renders correctly', () => {
  it('Renders without errors', () => {
    const wrapper = setup();
    expect(wrapper.length).toBe(1);
  });

  it('Renders correct username greeting', () => {
    const props = { username: 'user' };
    const wrapper = setup(props);
    const user = findByDataTest(wrapper, 'username');
    expect(user.length).toBe(1);
  });

  it('Renders correct generic greeting (no username)', () => {
    const wrapper = setup();
    const user = findByDataTest(wrapper, 'username');
    expect(user.length).toBe(0);
  });
});

describe('Greeting component has correct props', () => {
  it('Should not have any prop errors', () => {
    const expectedProps = {
      username: 'test',
      timeGreeting: 'morning'
    };
    const propsError = checkProps(Greeting, expectedProps);
    expect(propsError).toBeUndefined();
  });
});
