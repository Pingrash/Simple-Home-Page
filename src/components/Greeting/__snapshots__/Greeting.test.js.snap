// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting snapshot checks Renders correctly with username 1`] = `
<div
  className="greeting-container"
  data-test="container"
>
  <h3
    className="greeting-body"
    data-test="greeting"
  >
    Good 
    morning
    <span
      data-test="username"
    >
       
      user
      !
    </span>
  </h3>
</div>
`;

exports[`Greeting snapshot checks Renders correctly without username 1`] = `
<div
  className="greeting-container"
  data-test="container"
>
  <h3
    className="greeting-body"
    data-test="greeting"
  >
    Good 
    morning
    !
  </h3>
</div>
`;
